datalad/interface/save.py

```diff
@@ -81,6 +83,26 @@ def __call__(message=None, files=None, dataset=None,
         if auto_add_changes:
             files = [ds.path]
 
+        # treat special case of still untracked subdatasets.
+        # those need to become submodules now, as they are otherwise added
+        # without an entry in .gitmodules, and subsequently break Git's
+        # submodule functionality completely
+        for utf in ds.repo.repo.untracked_files:
+            utf_abspath = opj(ds.path, utf)
+            if not isdir(utf_abspath):
+                # this cannot be a repository
+                continue
+
+            # test whether the potential submodule is scheduled for saving
+            if sum([realpath(utf_abspath).startswith(_with_sep(realpath(f)))
+                    for f in files]):
+                # matches at least one path -> turn into submodule
+                _install_subds_inplace(
+                    ds=ds,
+                    path=utf_abspath,  # can be ignored, we don't need the return value
+                    relativepath=utf,
+                    name=None)
```

Comment:
so what happens if that submodule wasn't listed among files?  then old behavior? (`git add .` and having it added as a SubProject by git but not as a proper submodule?)